### PR TITLE
Update MassSpecFractionsDomainKind with changes to SampleSetDomainKind

### PR DESCRIPTION
**Pull request by Sigmonia**

### CODE REVIEW

File: ms2/src/org/labkey/ms2/metadata/MassSpecFractionsDomainKind.java [hunk 1]
```diff
@@ -122,8 +125,19 @@ public void appendNavTrail(NavTree root, Container c, User user)
     }
 
     @Override
-    public Domain createDomain(GWTDomain domain, JSONObject arguments, Container container, User user, @Nullable TemplateInfo templateInfo)
+    public Domain createDomain(GWTDomain domain, SampleTypeDomainKindProperties arguments, Container container, User user, @Nullable TemplateInfo templateInfo)
     {
-        return _assayDelegate.createDomain(domain, arguments, container, user, templateInfo);
+        //TODO this is probably broken.
```
Comment (cnathe) [inline ms2/src/org/labkey/ms2/metadata/MassSpecFractionsDomainKind.java]: @labkey-ians Is this TODOs still valid or have you verified these changes? 

@labkey-jeckels Josh, can you review this PR as it relates to the PR changes that Binal made for the other domain kinds?

File: ms2/src/org/labkey/ms2/metadata/MassSpecFractionsDomainKind.java [hunk 2]
```diff
@@ -122,8 +125,19 @@ public void appendNavTrail(NavTree root, Container c, User user)
...
+        JSONObject args = arguments != null ? arguments.toJSONObject() : null;
+        return _assayDelegate.createDomain(domain, args, container, user, templateInfo);
+    }
+
+    @Override
+    @NotNull
+    public ValidationException updateDomain(GWTDomain<? extends GWTPropertyDescriptor> original, GWTDomain<? extends GWTPropertyDescriptor> update, @Nullable SampleTypeDomainKindProperties options, Container container, User user, boolean includeWarnings)
+    {
+        //TODO this is probably broken too. Maybe we should throw a not supported exception here?
```
Comment (cnathe) [inline ms2/src/org/labkey/ms2/metadata/MassSpecFractionsDomainKind.java]: same comment as above related to this TODO

**Review by labkey-jeckels — APPROVED** [review]
I pulled mothership reporting data and have started a campaign to delete this unused assay type.

Since there's some test coverage (XtandemTest and others) and they're passing in the FB, I'm OK with merging this, though I suggest updating the comments.

File: ms2/src/org/labkey/ms2/metadata/MassSpecFractionsDomainKind.java (hunk 2 above)
Comment (Sigmonia) [reply ms2/src/org/labkey/ms2/metadata/MassSpecFractionsDomainKind.java]: This method isn't called as far as I can tell. This overrides the underlying SampleSetDomainKind, but the implemented Assay gets updated through the AssayService. The better option here I think is throwing a NotImplementedException or similar.

**Pull request merged**